src/stream.rs

```diff
@@ -1,101 +1,205 @@
 use std::io::{Read, Seek};
-use std::sync::{Arc, Weak};
+use std::marker::Sync;
+use std::sync::Arc;
 use std::{error, fmt};
 
 use crate::decoder;
-use crate::dynamic_mixer::{self, DynamicMixerController};
+use crate::dynamic_mixer::{mixer, Mixer, MixerSource};
 use crate::sink::Sink;
-use crate::source::Source;
 use cpal::traits::{DeviceTrait, HostTrait, StreamTrait};
-use cpal::{Sample, SupportedStreamConfig};
+use cpal::{
+    BufferSize, ChannelCount, FrameCount, Sample, SampleFormat, SampleRate, StreamConfig,
+    SupportedBufferSize,
+};
 
-/// `cpal::Stream` container. Also see the more useful `OutputStreamHandle`.
+const HZ_44100: cpal::SampleRate = cpal::SampleRate(44_100);
+
+/// `cpal::Stream` container. Use `mixer()` method to control output.
 ///
-/// If this is dropped playback will end & attached `OutputStreamHandle`s will no longer work.
+/// If this is dropped, playback will end, and the associated output stream will be disposed.
 pub struct OutputStream {
-    mixer: Arc<DynamicMixerController<f32>>,
+    mixer: Arc<Mixer<f32>>,
     _stream: cpal::Stream,
 }
 
-/// More flexible handle to a `OutputStream` that provides playback.
-#[derive(Clone)]
-pub struct OutputStreamHandle {
-    mixer: Weak<DynamicMixerController<f32>>,
+impl OutputStream {
+    pub fn mixer(&self) -> Arc<Mixer<f32>> {
+        self.mixer.clone()
+    }
 }
 
-impl OutputStream {
-    /// Returns a new stream & handle using the given output device and the default output
-    /// configuration.
-    pub fn try_from_device(
-        device: &cpal::Device,
-    ) -> Result<(Self, OutputStreamHandle), StreamError> {
+#[derive(Copy, Clone, Debug)]
+pub struct OutputStreamConfig {
+    pub channel_count: ChannelCount,
+    pub sample_rate: SampleRate,
+    pub buffer_size: BufferSize,
+    pub sample_format: SampleFormat,
+}
```

Comment:
The motivation of making this a separate struct, was to let users to pass it around or keep it in configs somewhere, for example. Yes, there is no problem to not publish it just yet.

Reply:
If I understand correctly `OutputStreamConfig` is public to allow users to skip using the builder a second time. 

One of the big advantages to builders is that we can expand them over time without that being a breaking change. 

Further more the `with_supported_config` implies the config will always be valid for the duration of the program. An audio device can disconnect however (for example usb dac/headphones). A config stored earlier might not be supported anymore.

My advice is to remove `OutputStreamBuilder` completely from the public API, and also remove `with_supported_config`. Users wanting to abstract away the builder they can do that in their own project.

Reply:
Usually I prefer plain data if possible. But in this case I have allowed this option (to use the config directly)  just to cover all the cases. But since no one actually requested this,  I do not object hiding the struct.

Updated.

Reply:
`OutputStreamBuilder` fields no longer need to be public (it compiles with pub removed). 

Sorry I would apply the change to the PR myself but its such a chore to set that up. Ill see if I can automate that better on my side so its no longer a problem in the future.

Reply:
you know what ill fix it post merge, cause this is all ready to go! :partying_face: